torch/_higher_order_ops/while_loop.py

```diff
@@ -216,17 +239,22 @@ def while_loop_func(ctx, cond_fn, body_fn, operands):
             (functional_body_fn, "body_fn"),
         ]:
             if _has_potential_branch_input_mutation(
-                fn, unwrapped_operands, pre_dispatch=pre_dispatch
+                fn, unwrapped_inputs, pre_dispatch=pre_dispatch
             ):
                 raise UnsupportedAliasMutationException(
                     f"torch.while_loop's {fn_name} might be modifying the input!"
                 )
 
             if _has_potential_branch_input_alias(
-                fn, unwrapped_operands, pre_dispatch=pre_dispatch
+                fn, unwrapped_inputs, pre_dispatch=pre_dispatch
             ):
                 raise UnsupportedAliasMutationException(
                     f"torch.while_loop's {fn_name} might be aliasing the input!"
                 )
-        ret = while_loop_op(functional_cond_fn, functional_body_fn, unwrapped_operands)
+        ret = while_loop_op(
+            functional_cond_fn,
+            functional_body_fn,
+            unwrapped_carried_inputs,
+            unwrapped_additional_inputs,
```

Comment:
Do we allow non-Tensor inputs in any of those?

Reply:
Sorry, i was wrong. We've allowed tensor, and python int, float, bool to be passed in as inputs. Maybe we should add more tests for them.

Reply:
Inductor support for `torch.while_loop` currently only covers Tensor inputs (carried and additional), as we fetch fake tensors for them and pass those to subgraph lowering. We'll need special treatment of non-Tensor inputs.

Reply:
We could defer it until a valid test cases I think.